core/src/sync/state/snapshot_chunk_sync.rs

```diff
@@ -252,8 +259,68 @@ impl SnapshotChunkSync {
                 "Snapshot chunks are all downloaded in {:?}",
                 download_start_time.elapsed()
             );
+            // TODO: these code should be called after restore completed
```

Comment:
From the comment, it seems that "Checking blame state" is a new phase immediately after the "Sync checkpoint" phase.